### ISSUE TITLE
Setting value of MultiSelect in javascript doesn't unselect previous selected items.

### ISSUE THREAD

**Issue by ruoyu0088** [bokeh/bokeh#7926]
Here is the demo code, by clicking button A, B, C in sequence, the value of `select.value` is correct: `["C"]`, but all the items are selected.

```
from bokeh.plotting import figure, output_notebook, show
from bokeh.models import ColumnDataSource, CustomJS, MultiSelect, Button
from bokeh.layouts import row, column
output_notebook()

select = MultiSelect(options=["A", "B", "C"])
buttons = [Button(label=label) for label in select.options]

def callback_func(select=select):
    select.value = [cb_obj.label]
    print(select.value)

callback = CustomJS.from_py_func(callback_func)    
    
for button in buttons:
    button.callback = callback
    
show(column(select, row(buttons)))
```

The problem is the same when setting value  in Python , Here is the server version:

```
def app(doc):
    select = MultiSelect(options=["A", "B", "C"])
    buttons = [Button(label=label) for label in select.options]
    
    def set_label(label):
        select.value = [label]
        
    for button in buttons:
        button.on_click(lambda label=button.label:set_label(label))

    doc.add_root(column(select, row(buttons)))

show(app)
```

![firefox_2018-05-24_11-35-06](https://user-images.githubusercontent.com/826871/40461499-a0f28cc8-5f46-11e8-94a4-bf2459a7f9d9.png)

**Comment by bryevdv** [comment]
`CustomJS` callbacks only affect *JavaScript state*. There is no update of any Python variable or state. For that kind of connection you must use a Bokeh server app (that is the purpose of the Bokeh server) http://bokeh.pydata.org/en/latest/docs/user_guide/server.html

**Comment by bryevdv** [comment]
Sorry, the wording of your issue was confusing to me:

> but the value of select.value is `["C"]`

makes it sound like your unmet expected was about the value, rather than the visual presentation.

@mattpap The following diff seems to fix things outside the notebook:
```
diff --git a/bokehjs/src/lib/models/widgets/multiselect.ts b/bokehjs/src/lib/models/widgets/multiselect.ts
index bfc07a355..c3120e5f0 100644
--- a/bokehjs/src/lib/models/widgets/multiselect.ts
+++ b/bokehjs/src/lib/models/widgets/multiselect.ts
@@ -62,6 +62,8 @@ export class MultiSelectView extends InputWidgetView {
     for (const el of Array.from(this.el.querySelectorAll('option'))) {
       if (values[el.value])
         el.selected = true
+      else
+        el.selected = false
     }

     // Note that some browser implementations might not reduce
```
but I haven't been able to confirm this fixes in the notebook because of typical notebook garbage.

**Comment by bryevdv** [comment]
Thanks, I'll let you make the PR.

**Comment by mattpap** [comment]
> `el.selected = values[el.value]`

Actually this is incorrect, because it would rely on implicit conversions. In fact, it wouldn't even compile. The correct code is `el.selected = el.value in values`. However, this raises a question, why use map for `values`, where a set would be much a much more appropriate data structure. I included this simplification in PR #7932.
